src/Components/Aspire.Npgsql.EntityFrameworkCore.PostgreSQL/AspireEFPostgreSqlExtensions.cs

```diff
@@ -117,12 +121,52 @@ void ConfigureDbContext(DbContextOptionsBuilder dbContextOptionsBuilder)
 
         void ConfigureRetry()
         {
-            if (!settings.Retry)
+#pragma warning disable EF1001 // Internal EF Core API usage.
+            if (settings.Retry || settings.CommandTimeout.HasValue)
             {
-                return;
+                builder.PatchServiceDescriptor<TContext>(optionsBuilder => optionsBuilder.UseNpgsql(options =>
+                {
+                    var extension = optionsBuilder.Options.FindExtension<NpgsqlOptionsExtension>();
+
+                    if (settings.Retry)
+                    {
+                        var executionStrategy = extension?.ExecutionStrategyFactory?.Invoke(new ExecutionStrategyDependencies(null!, optionsBuilder.Options, null!));
+
+                        if (executionStrategy != null)
+                        {
+                            if (executionStrategy.GetType() == typeof(NpgsqlRetryingExecutionStrategy))
+                            {
+                                // Keep custom Retry strategy
+                            }
+                            else if (executionStrategy.GetType() != typeof(NpgsqlExecutionStrategy))
+                            {
+                                throw new InvalidOperationException($"{nameof(NpgsqlEntityFrameworkCorePostgreSQLSettings)}.Retry can't be set when a custom Execution Strategy is configured.");
+                            }
```

Comment:
Can't use is?

Reply:
You don't want to use `is` because that would return `true` for that type and any derived type. Here we are saying "if the executionStrategy is concretely an NpgsqlExecutionStrategy, we will call `EnableRetryOnFailure()`. For anything else, throw"

Reply:
Got it. A comment would be great 😄

Reply:
You're saying you specifically want to exclude subclasses of e.g. NpgsqlRetryingExecutionStrategy? I think it makes sense to assume that such subclasses would also be retrying (otherwise they wouldn't/shouldn't derive from NpgsqlRetryingExecutionStrategy), but ok...

Reply:
Good point @roji. I think Line 137 should be ` if (executionStrategy is NpgsqlRetryingExecutionStrategy)`.

Line 141 should still be `else if (executionStrategy.GetType() != typeof(NpgsqlExecutionStrategy))`.

Reply:
Done